haproxy/README.md

```diff
@@ -12,9 +12,17 @@ Capture HAProxy activity in Datadog to:
 
 ## Setup
 
+### Prerequisites
+
+This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version (enterprise version 1.9rc1 or later) to enable the Prometheus endpoint on HAProxy. If you are using the legacy version, use the [HAProxy Prometheus exporter][3], or set up the [socket-based integration](#using-the-stats-endpoint). For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][29].
```

Comment:
```suggestion
### Prerequisites

This OpenMetrics-based integration has a latest version (V2) and a legacy version (V1). To get all the most up-to-date features, Datadog recommends upgrading to the latest version (enterprise version 1.9rc1 or later) to enable the Prometheus endpoint on HAProxy. If you are using the legacy version, use the [HAProxy Prometheus exporter][3], or set up the [socket-based integration](#using-the-stats-endpoint). For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][29].
```

Reply:
My take:

```suggestion
This integration can collect metrics from a Prometheus endpoint via (recommended) or via a socket-based integration through the stats endpoint (deprecated). Using the Prometheus endpoint requires HAProxy version 2 (enterprise version 1.9rc1) or later.

When using the Prometheus endpoint, starting with version 4.0.0, this OpenMetrics-based integration has a *latest* mode (`use_openmetrics`: true) and a *legacy* mode (`use_openmetrics`: false and `use_prometheus`: true). To get all the most up-to-date features, Datadog recommends enabling the *latest* mode. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][29].

To use the socket-based integration, set both `use_openmetrics` and `use_prometheus` to false and follow the [corresponding instructions](#using-the-stats-endpoint) on the Configuration section.
```

I know it's confusing because of the two layers of legacy that we have on this integration. Hopefully this suggestion addresses that, and helps somewhat with what #15449 addresses. Some of the information becomes redundant with things that are also stated in the configuration section, and I'm not sure how to address that, but I feel that the global view of the "first layer of legacy" needs to be explained this time before getting into the OpenMetrics legacy vs latest stuff.